### PR TITLE
Fix 17786 - CompileBefore/CompileAfter is lost due to the FSharpSourceCodeCompileOrder target

### PR DESCRIPTION
**Pull request by KevinRansom**
Fixes: #17786 - CompileBefore/CompileAfter is lost due to the FSharpSourceCodeCompileOrder target

This adds the CompilationOrder metadata as default metadata to CompileBefore/CompileAfter and CompileLast.  This ensures that when they are inserted into the Compile/Sources collection the metadata goes with them.

Currently:
image

With this change:
image

### CODE REVIEW

**Comment by KevinRansom** [comment]
@auduchinok let me know if this works for you.

**Review by T-Gro — APPROVED** [review]

File: src/FSharp.Build/Microsoft.FSharp.NetSdk.props [hunk 1]
```diff
@@ -130,6 +130,18 @@ WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and
...
+      <CompileOrder>CompileBefore</CompileOrder>
+    </CompileBefore>
+
+    <CompileAfter>
+      <CompileOrder>CompileAfter</CompileOrder>
+    </CompileAfter>
+
+    <CompileLast>
+      <CompileOrder>CompileLast</CompileOrder>
+    </CompileLast>
```
Comment (auduchinok) [inline src/FSharp.Build/Microsoft.FSharp.NetSdk.props:144]: Could you please also add `CompileFirst`?

I see its usage in `FSharpSourceCodeCompileOrder` here: https://github.com/dotnet/fsharp/blob/6cdcc3c33ca162988f3f90033f891bc265b9a3bc/src/FSharp.Build/Microsoft.FSharp.Targets#L262

and in `FSharp.Core` here: https://github.com/dotnet/fsharp/blob/6cdcc3c33ca162988f3f90033f891bc265b9a3bc/src/FSharp.Core/FSharp.Core.fsproj#L80

**Review by auduchinok — APPROVED** [review]
@KevinRansom Wow, this is such a simple fix! Looks great, thank you very much! 🙂

File: src/FSharp.Build/Microsoft.FSharp.NetSdk.props (hunk 1 above)
Comment (auduchinok) [reply src/FSharp.Build/Microsoft.FSharp.NetSdk.props:144]: @KevinRansom This hasn't been resolved. 😞

**Pull request merged**